reconstruction/rtpc/src/main/java/org/jlab/rec/rtpc/hit/KF/KalmanFilter.java

```diff
@@ -0,0 +1,472 @@
+
+                if (!(sitemap == null)) if (!sitemap.isEmpty()) sitemap.clear();
+                if (!(statemap == null)) if (!statemap.isEmpty()) statemap.clear();
+                if (!(matrixmap == null)) if (!matrixmap.isEmpty()) matrixmap.clear();
+                if (!(bstatemap == null)) if (!bstatemap.isEmpty()) bstatemap.clear();
+                if (!(bmatrixmap == null)) if (!bmatrixmap.isEmpty()) bmatrixmap.clear();
+
+        }
+
+        params.set_kftrackinfomap(kftrackinfomap);
+    }
+
+    private void test3() throws Exception{
+        boolean debug = false;
+
+        State state = statemap.get(0);
+        SimpleMatrix P = matrixmap.get(0);
+        if (print) state.debug("start", sitemap.get(0));
+
+        Group target = truc(state, sitemap.get(0), sitemap.get(1), new Material("Deuterium"), true, debug);
+        Group wall = truc(target.state, sitemap.get(1), sitemap.get(2), new Material("Kapton"), true, debug);
+        Group buffer = truc(wall.state, sitemap.get(2), sitemap.get(3), new Material("HeCO2"), true, debug);
+        Group foil = truc(buffer.state, sitemap.get(3), sitemap.get(4), new Material("Mylar"), true, debug);
+        Group flow = truc(foil.state, sitemap.get(4), sitemap.get(5), new Material("HeCO2"), true, debug);
+        Group cath = truc(flow.state, sitemap.get(5), sitemap.get(6), new Material("Mylar"), true, debug);
+        Group meas = truc(cath.state, sitemap.get(6), sitemap.get(7), new Material("HeCO2"), true, debug);
+
+        State state_minus = meas.state;
+        SimpleMatrix F = target.F.mult(wall.F).mult(buffer.F).mult(foil.F).mult(flow.F).mult(cath.F).mult(meas.F);
+        SimpleMatrix Q = target.Q.plus(wall.Q).plus(buffer.Q).plus(foil.Q).plus(flow.Q).plus(cath.Q).plus(meas.Q);
+
+        SimpleMatrix P_minus = (F.mult(P).mult(F.transpose())).plus(Q);
+
+        Site to = sitemap.get(7);
+        SimpleMatrix h = state_minus.h(to);
+        SimpleMatrix m = to.get_m();
+        SimpleMatrix y = m.minus(h);
+        y.set(1, 0, modPI(y.get(1, 0)));
+
+        SimpleMatrix H = state_minus.H(to);
+        SimpleMatrix R = to.R();
+        SimpleMatrix S = H.mult(P_minus).mult(H.transpose()).plus(R);
+
+        SimpleMatrix K = P_minus.mult(H.transpose().mult(S.invert()));
+
+        SimpleMatrix a_plus = state_minus.get_sv().plus(K.mult(y));
+        SimpleMatrix P_plus = P_minus.minus(K.mult(H).mult(P_minus));
+
+        State sv_plus = new State(a_plus, 0);
+
+        sv_plus.updatePhi(to);
+
+        if (print) sv_plus.debug("update", to);
+
+        statemap.put(7, sv_plus);
+        matrixmap.put(7, P_plus);
+
+        for (int k = 7; k < sitemap.size() - 1; k++) {
+            truc2(k, debug);
+        }
+
+        fillbMap();
+
+        // Back Propagation !!!!
+        for (int k = sitemap.size() - 1; k > 7; k--) {
+            truc3(k, debug);
+        }
+        lasttruc(debug);
+
+
+    }
+
+    private void truc2(int k, boolean debug) throws Exception {
```

Comment:
Can you change the name of this method to something more representative of what it does? transportAndFilter?